### ISSUE TITLE
[gha][cluster-test] unify error reporting

### ISSUE THREAD

**Issue by rustielin** [libra/libra#5122]
After https://github.com/libra/libra/pull/5096, we now log errors and metrics even on experiment suite failure.

This lets us have something like: 

---

Cluster Test Result
```
Experiment `all up` failed: `Failed to run experiment: TEST INDUCE FAILURE!!!`
```

 Cluster Test failed - test report processing failed. See https://github.com/libra/libra/actions/runs/170439368
Repro cmd:

    ./scripts/cti --tag land_40381b6c --run bench

---

Test failure at this moment occurs not when `cti` itself fails, but by analyzing the report that it generates (or lack of). We should check for error by exit code of `cti` rather than trying to collect metrics and catching if that fails.
- [x] Check exit code of `cti` and do error handling on that in LBT
- [x] Add more helpful errors and report text experiments, particularly `compatibility_test`, as those will be reflected in PR comment by bors

For the latter, probably something specific at each experiment phase:

```
// check storage
// pass storage check

context.report.report_text("2. Changing images for the first batch to test consensus".to_string());
update_batch_instance(context, &self.first_batch, self.updated_image_tag.clone()).await?;
context
    .tx_emitter
    .emit_txn_for(job_duration, validator_txn_job.clone())
    .await
    .map_err(|e| anyhow::format_err!("Storage compatible, but consensus backwards compat broken: {}", e))?;
```

**Comment by rustielin** [comment]
For checking `cti` exit code:
* Error: https://github.com/libra/libra/runs/875428478
* Success: https://github.com/libra/libra/runs/875478836

**Comment by rustielin** [comment]
resolved by #5150
